### PR TITLE
Fix/dupes

### LINKED ISSUE #2136: Duplication of villager
### Minecolonies version

Forge 2604 ( in modpack )
version ; minecolonies-universal-1.12.2-0.8.7273
- Barbarian is disabled in config file

### Expected behavior

- No duplication

### Actual behaviour

- Trace of duplication.

1 - In fml-junk-earlystartup.log ( extract )
[fml-junk-earlystartup.log](https://github.com/Minecolonies/minecolonies/files/1688417/fml-junk-earlystartup.log)

2 - In player console.
villager

3 - Old picture in old post.

### Steps to reproduce the problem

Not found a process for reproduct at this time.

### LINKED ISSUE #2438: NPC's duplicate when you leave the area then come back
minecolonies-universal-1.12.2-0.8.8208

i made a builder hut and whenever I leave the area so the chunks unload then come back there is one standing where it last was and a new one duplicated. it can keep happening so there can be infinite amounts of one guy till you re-call it individually

### LINKED ISSUE #2443: Warehouse reposition not possible due to wrong condition
### Minecolonies version

version 1.12.2-0.8.8208

### Expected behavior

- warehouse reposition is possible

### Actual behaviour

- chat message: "The amount of warehouses is limited to 1 per colony."

### Steps to reproduce the problem

1. Build level 1 warehouse.
2. Try to reposition.
3. Profitless bug

BTW: reposition seems to be very buggy sometimes - found many build orders in the town hall which I didn't issued (instead of one reposition which I ordered) causing two big holes under a building (level 4 builder's hut), solved by destroying whole building, firing/hiring again the same builder and repairing the building

### LINKED ISSUE #2455: Re-positioning town hall deletes the podium
### Minecolonies version

version latest

### Expected behavior

- will place podium back down

### Actual behaviour

- deletes the podium saying you can not have more than 1 town hall

### Steps to reproduce the problem

1. XYZ
2. ZYX
3. ...
4. Profit

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2136 
Closes #2438 
Closes #2443 
Closes #2455 

# Changes proposed in this pull request:
- Should fix colonist duplication and vanishing.
- Also should fix building moving of warehouse and townhall.
- Also reduces log spam.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java [hunk 1]
```diff
@@ -106,26 +107,41 @@ public void sendPackets(
...
             {
-                if (citizen.isDirty() || hasNewSubscribers)
+                if (citizen.getCitizenEntity().isPresent())
                 {
-                    if (citizen.getCitizenEntity().isPresent())
+                    final List<EntityCitizen> list = colony.getWorld()
+                            .getEntities(EntityCitizen.class,
+                                    entityCitizen -> entityCitizen.getColony().getID() == colony.getID() && entityCitizen.getCitizenData().getId() == citizen.getId());
+
+                    if (!list.isEmpty() && citizen.getCitizenEntity().get().getEntityId() != list.get(0).getEntityId())
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java:116]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 95 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 83 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L216](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L216): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L217](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L217): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L218](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L218): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L219](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L219): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L220](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L220): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L221](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L221): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L854](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L854): Use already-defined constant 'TAG_INVENTORY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/ee5491806594d0b82e8d46917b2c7c130ec9499b/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Nightenom** [comment]
Should also close #2455

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 90 issues
* ![CRITICAL][CRITICAL] 7 critical
* ![MAJOR][MAJOR] 81 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [CitizenData.java#L216](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L216): Use already-defined constant 'TAG_SKILLS' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L217](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L217): Use already-defined constant 'TAG_SKILL_STRENGTH' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L218](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L218): Use already-defined constant 'TAG_SKILL_STAMINA' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L219](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L219): Use already-defined constant 'TAG_SKILL_SPEED' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L220](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L220): Use already-defined constant 'TAG_SKILL_INTELLIGENCE' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L221](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L221): Use already-defined constant 'TAG_SKILL_DEXTERITY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [CitizenData.java#L854](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/colony/CitizenData.java#L854): Use already-defined constant 'TAG_INVENTORY' instead of duplicating its value here. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/165225d0de241b42aa84b45f50299684a488be0f/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**